django/forms/widgets.py

```diff
@@ -1023,7 +1023,7 @@ def format_value(self, value):
         31 to display correctly.
         """
         year, month, day = None, None, None
-        if isinstance(value, (datetime.date, datetime.datetime)):
+        if isinstance(value, datetime.date):
```

Comment:
I know that `datetime` is redundant in instance checks, but it's not obvious that `datetime` is a subclass of `date` and removing it makes the code less readable (at least for me). I've scratched my head multiple times in the past trying to fix issues related with different branches for `date` and `datetime`.

Reply:
Fair enough. I'll drop the first and the third commit here. (I found three other places that don't use `datetime.datetime` in addition to `datetime.date` in `isinstance()` checks, but one of them I'd only expect a `date` anyway.)

The second one should be fixed though as custom subclasses of `datetime.date` don't work if `type(...) is ...` is used. Will open a ticket for that later.